errdef/errors.go

```diff
@@ -27,4 +27,5 @@ var (
 	ErrSizeExceedsLimit   = errors.New("size exceeds limit")
 	ErrUnsupported        = errors.New("unsupported")
 	ErrUnsupportedVersion = errors.New("unsupported version")
+	ErrForeignLayer       = errors.New("foreign layer is not supported")
```

Comment:
Need advice for a better message that matches the error level.

Reply:
Changed accordingly, after offline discussion.

Reply:
I'm wondering would this specific `ErrForeignLayer` be checked by consumers (such as oras and notation)? If not, maybe we can just reuse `ErrUnsupported`.